test/cppsuite/CMakeLists.txt

```diff
@@ -54,6 +54,18 @@ create_test_executable(run
     CXX
 )
 
+create_test_executable(csuite_style_example_test
```

Comment:
Hi @alisonfel, we are trying to add a new test to this PR. I have updated the CMake build and it seems to work. However, `csuite_style_example_test` shares some source files with the `run` executable just above. We may have more and more tests similar to `csuite_style_example_test`. Is there a way to have "common source files" or a cleaner way to do things? Also, can you review the CMake changes, would be awesome! 🙏🏻

Reply:
Hmm, is there a minimum set of sources that will always remain constant between the the two tests? (i.e. needed to compile the `run`, `csuite_style_example_test` and future test definitions). Or are the sources being defined under each test definition uniquely based on the dependencies for that given test. 

If the former (i.e. a set of sources that will be used by all test definitions), you could easily define an intermediate variable to capture the sources e.g.
```
set (common_sources 
        test_harness/connection_manager.cxx
        test_harness/thread_manager.cxx
        test_harness/util/logger.cxx
        test_harness/util/scoped_types.cxx
        test_harness/workload/random_generator.cxx)

create_test_executable(csuite_style_example_test
     SOURCES ${common_sources} tests/csuite_style_example_test.cxx
     FLAGS -std=c++11
     INCLUDES ${CMAKE_CURRENT_SOURCE_DIR}
     CXX
```
If the latter (i.e source requirements are unique), the only other way I can think of besides your current approach (which I think is also perfectly fine) could be defining an intermediate static library to hold the test harness sources. e.g.
```
add_library(cppsuite_test_harness STATIC 
    test_harness/core/component.cxx 
    test_harness/core/configuration.cxx 
    test_harness/core/throttle.cxx 
    test_harness/util/logger.cxx
    test_harness/util/scoped_types.cxx 
    test_harness/workload/database_model.cxx 
    test_harness/workload/database_operation.cxx 
    test_harness/workload/random_generator.cxx 
    test_harness/workload/thread_context.cxx 
    test_harness/workload/workload_tracking.cxx 
    test_harness/workload/workload_validation.cxx 
    test_harness/checkpoint_manager.cxx 
    test_harness/connection_manager.cxx 
    test_harness/runtime_monitor.cxx 
    test_harness/test.cxx 
    test_harness/thread_manager.cxx 
    test_harness/timestamp_manager.cxx 
    test_harness/workload_generator.cxx 
)
target_include_directories( test_harness PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
target_compile_options( test_harness
    PUBLIC -std=c++11
)
create_test_executable(run 
    SOURCES  tests/run.cxx
    LIBS cppsuite_test_harness
    FLAGS -std=c++11
    CXX
)
create_test_executable(csuite_style_example_test 
    SOURCES  tests/csuite_style_example_test.cxx
    LIBS cppsuite_test_harness
    FLAGS -std=c++11
    CXX
)
```  
I haven't tested the above, but I think that should give a rough starting point to get it eventually working. An internal static library can be a useful tool to define a set of common sources + saves the cost of compiling the same object multiple times (we do this with the `test_util` library for our csuite tests). On the downside that `csuite_style_example_test` will include test_harness symbols that it may not end up using.

Reply:
Thank you for those two great suggestions !

Reply:
As discussed, I went for the second solution, see 637ca4cdab52d4128aa88ac267dbb728ee891ab8. Thanks for your help !